Symfony/CS/Error/Error.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Error;
+
+/**
+ * An abstraction for errors that can occur before and during fixing.
+ *
+ * @author Andreas Möller <am@localheinz.com>
+ *
+ * @internal
+ */
+class Error
+{
+    const TYPE_LINTING = 1;
+    const TYPE_FIXING = 2;
```

Comment:
nomenclature from FixerFileProcessedEvent:
- `INVALID` for invalid file (with syntaxes) input (before linting)
- `EXCEPTION` for unknown exception (during linting)
- `LINT` for exception (after fixing)

How about using it here?
Then, just map first one as external/user error, and 2nd+3rd as internal/tool error.